preprocessing.py

```diff
@@ -261,6 +261,40 @@ def get_tpu_bt_input_tensors(games, games_nr, batch_size, num_repeats=1,
     return dataset
 
 
+def get_many_tpu_bt_input_tensors(games, games_nr, batch_size,
+                                       start_at, num_datasets,
+                                       moves=2**21,
+                                       window_size=500e3,
+                                       window_increment=25000):
+    dataset = None
+    for i in range(num_datasets):
+        # TODO(amj) mixin calibration games with some math. (from start_at that
+        # is proportionally along compared to last_game_number?  comparing
+        # timestamps?)
+        ds = games.moves_from_games(start_at + (i * window_increment),
+                                       start_at + (i * window_increment) + window_size,
+                                       moves=moves,
+                                       shuffle=True,
+                                       column_family=bigtable_input.TFEXAMPLE,
+                                       column='example')
+        ds = ds.repeat(1)
+        ds = ds.map(lambda row_name, s: s)
+        dataset = dataset.concatenate(ds) if dataset else ds
```

Comment:
Regarding the general approach: if the training loop does multiple scans, I would expect to create a new dataset for each pass, rather than try to create a single enormous dataset, which I imagine would be harder to debug, inspect, etc.

Reply:
yes, but multiple calls to tpuestimator.train will create new graphs :(  I am not sure what a good solution for lazy evaluating of these Datasets would be.  As it is, it takes a real long time to build the datasets before training even starts -- i suspect the concatenate is doing something bad as things get slower and slower.